### PR TITLE
DBTP-387 Improve developer experience

### PR DESCRIPTION
**Pull request by WillGibson**
Changes made as part of migration to DBT PaaS work to help a person understand what the application is about and get going with it.

### CODE REVIEW

File: config/sample_env
Comment (WillGibson) [inline config/sample_env:1]: Replaced by `.env.sample`.

File: core/admin_views.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def admin_login_view(request):
 
     if request.user.is_authenticated:
         if not request.user.is_staff:
-            raise PermissionDenied
+            raise PermissionDenied('User is not staff')
```
Comment (WillGibson) [inline core/admin_views.py:27]: This would have saved me some time.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	docker-compose down
+
+migrate:
+	$(web) python manage.py migrate
+
+shell:
+	$(web) python manage.py shell
+
+up:
+	make migrate
```
Comment (yusufsheiqh) [inline Makefile:29]:
```suggestion
	$(MAKE) migrate
```

File: README.md [hunk 3]
```diff
@@ -1 +1,48 @@
...
+
+If you need to edit their user, make them a superuser etc., this can only be done after their user record has been created.
+
+## Working on the application
+
+### Git branching strategy
+
+This repository uses [Git Flow](https://www.gitkraken.com/learn/git/git-flow).
+
+However, paketobuildpacks/builder can't deal with a / in branch names, so no `feature/...` etc. please.
```
Comment (yusufsheiqh) [inline README.md:24]:
```suggestion
However, `paketobuildpacks/builder` can't deal with slashes (`/`) in branch names, so no `feature/...` etc. please.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,48 @@
...
+
+### Managing your shell's environment variables
+
+A good way to manage the ones you want in your shell is to use [direnv](https://direnv.net/) with a `.envrc` file based on the `.envrc.sample` file in the root of this repository.
+
+### Running the application locally with Docker Compose
+
+Make yourself a `.env` based on `.env.sample`
```
Comment (yusufsheiqh) [inline README.md:32]:
```suggestion
Make yourself a `.env` based on `.env.sample`: `cp .env.sample .env`
```

File: runtime.txt [hunk 5]
```diff
@@ -1 +1 @@
-python-3.10.4
+python-3.10.12
```
Comment (yusufsheiqh) [inline runtime.txt:1]: If you haven't done so already... worth checking if this version is supported by GOV.UK PaaS.

File: Makefile (hunk 2 above)
Comment (WillGibson) [reply Makefile:29]: Oh damn, I meant to do that!

File: runtime.txt (hunk 5 above)
Comment (yusufsheiqh) [reply runtime.txt:1]: I am not sure... you can check which buildpack is being used via `cf buildpacks`.

Comment (WillGibson) [reply runtime.txt:1]:
```
➜ cf buildpacks                
Getting buildpacks as will.gibson@digital.trade.gov.uk...

position   name                    stack        enabled   locked   filename
...
18         python_buildpack        cflinuxfs4   true      false    python-buildpack-cflinuxfs4-v1.8.13.zip
...
```

Should be OK for now, probably more likely to work that it was before 😆

**Review by yusufsheiqh — APPROVED** [review]

**Pull request merged**